extract.js

```diff
@@ -73,7 +73,7 @@ function generateFeatures(url, client, clientid) {
             if (features[fname].length === 2) {
                 let feature = features[fname].apply(null, [client, conn]);
                 if (feature !== undefined) {
-                    if (typeof feature === 'object' && Object.keys(feature).length > 0) {
+                    if (feature !== null && Object.keys(feature).length > 0) {
```

Comment:
I attempted to use `Object.keys(feature) || {}`, but looks like that doesn't fix the `null` issue:
```
> feature = null;
null
> if (Object.keys(feature)){ console.log("yes"); }
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
> if (Object.keys(feature) || {}){ console.log("yes"); }
TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
```

Reply:
I think what you want is `Object.keys(feature || {})`

Reply:
Won't this always evaluate to `true`?

Reply:
nevermind ... just pushed an update with improved `if` statement

Reply:
If you do keep the `typeof feature === 'object'` piece, it would protect against landing in this case if it was a string or array or whatever.

So it would be `typeof feature === 'object' && Object.keys(feature || {}).length > 0`

Or is that not necessary?

Reply:
I was thinking that the `typeof` check was actually implicit, but a string here would actually give an incorrect result. I re-added the `typeof` check.